common/server-mainloop.c

```diff
@@ -286,18 +220,18 @@ server_run()
             ASSERT(timcb->callback);
 
             /* Call any timers that have already passed */
-            if(timeval_compare(&current, &timcb->at) >= 0)
+            if(timercmp(&current, &timcb->at, >=))
             {
                 /* Convert to milliseconds, and make the call */
                 r = (timcb->callback)(timeval_to_ms(current), timcb->arg);
 
                 /* Reset timer if so desired */
-                if (r == 1 && !timeval_empty(&timcb->interval))
+                if (r == 1 && timerisset(&timcb->interval))
                 {
-                    timeval_add(&timcb->at, &timcb->interval);
+                    timeradd(&timcb->at, &timcb->interval, &timcb->at);
 
-                    /* If the time has already passed, just use current time */
-                    if(timeval_compare(&(timcb->at), &current) <= 0)
+                    /* If the new timeout has already passed, reset it to current time */
```

Comment:
Just a nit here - should the "timeout" here have said "timer" (or just "time") ?